apps/meteor/ee/app/livechat-enterprise/server/api/rooms.ts

```diff
@@ -1,56 +1,80 @@
-import { Meteor } from 'meteor/meteor';
-import { isPOSTLivechatRoomPriorityParams } from '@rocket.chat/rest-typings';
+import { isLivechatRoomOnHoldProps, isLivechatRoomResumeOnHoldProps, isPOSTLivechatRoomPriorityParams } from '@rocket.chat/rest-typings';
 import { LivechatRooms, Subscriptions } from '@rocket.chat/models';
+import type { IOmnichannelRoom } from '@rocket.chat/core-typings';
+import { OmnichannelEEService } from '@rocket.chat/core-services';
 
 import { API } from '../../../../../app/api/server';
 import { hasPermissionAsync } from '../../../../../app/authorization/server/functions/hasPermission';
-import { LivechatEnterprise } from '../lib/LivechatEnterprise';
 import { removePriorityFromRoom, updateRoomPriority } from './lib/priorities';
 import { i18n } from '../../../../../server/lib/i18n';
 
 API.v1.addRoute(
 	'livechat/room.onHold',
-	{ authRequired: true, permissionsRequired: ['on-hold-livechat-room'] },
+	{ authRequired: true, permissionsRequired: ['on-hold-livechat-room'], validateParams: isLivechatRoomOnHoldProps },
 	{
 		async post() {
 			const { roomId } = this.bodyParams;
-			if (!roomId || roomId.trim() === '') {
-				return API.v1.failure('Invalid room Id');
+			if (roomId.trim() === '') {
+				throw new Error('error-invalid-room');
```

Comment:
using the validator is possible to receive an empty string?

Reply:
Yups, you're right. This is not needed